dvc/remote/ssh.py

```diff
@@ -45,7 +45,10 @@ class RemoteSSH(RemoteBase):
     scheme = 'ssh'
 
     # NOTE: temporarily only absolute paths are allowed
-    REGEX = r'^ssh://((?P<user>.*)@)?(?P<host>[^/]*):(?P<path>/.*)$'
+    REGEX = (r'^ssh://((?P<user>.*)@)?'
```

Comment:
@efiop, maybe we don't even need a custom regex to understand the URLs, what about `urllib.parse`? 

The API should expose the parts of the URL we are interested in.
Also, `urlparse` understands almost everything from connection urls:
```python
from urllib.parse import urlparse

p = urlparse('postgresql://mroutis:changeme@localhost:8080/database')

p.schema   # postgresql
p.username # mroutis
p.password # changeme
p.hostname # localhost
p.port     # 8080
p.path     # /database
```

The only problem I'm currently seeing and that stopped me from an RFC on the remotes, is Azure, they use a different kind of URL format (more like several key=value statements joined with a semicolon, IIRC), in this case, I was thinking to delegate the "url" validation to the Azure module, but didn't get that far on their documentation.

I know that parsing them with a regex is usually not an issue and sometimes pretty straight forward, however we can delegate that work to the standard library and slim our remotes' logic a little bit, what do you think?

_(NOTE: This is a copy paste from the comment right here: https://github.com/iterative/dvc/issues/1473#issuecomment-451975839)_

Reply:
Overall, LGTM :+1:

Reply:
Great catch! I'll fix it to use urlparse in a sec. Could you take a look at https://github.com/iterative/dvc/pull/1470, please? There are some useful RFC things like moving urlparse imports to utils.py, that can be used here as well.

Reply:
@mroutis Also, please don't hesitate to use "Request changes" instead of "Approve" when reviewing, since this as well as other comments you've made should definitely be accounted for before merging, thus requireing changes :)